rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -912,11 +938,64 @@ void main_window::DecryptSPRXLibraries()
 				{
 					gui_log.error("Failed to create %s", new_path);
 				}
+
+				break;
 			}
-			else
+			else if (!tried)
 			{
-				gui_log.error("Failed to decrypt %s", old_path);
+				// Allow the user to manually type KLIC if decryption failed
+
+				const std::string filename = old_path.substr(old_path.find_last_of(fs::delim) + 1);
+
+				input_dialog dlg(32, QString{}, tr("Enter KLIC of %0").arg(qstr(filename)),
+					tr("Hexadecimal only."), "00000000000000000000000000000000", this);
+
+				QFont mono = QFontDatabase::systemFont(QFontDatabase::FixedFont);
+				mono.setPointSize(8);
+				dlg.edit_input_font(mono);
+				dlg.disable_clear_button();
+				dlg.set_validator(new QRegExpValidator(QRegExp("^[a-fA-F0-9]*$"))); // HEX only
+
+				QString result;
+				connect(&dlg, &input_dialog::text_changed, [&](const QString& text)
+				{
+					result = text;
+				});
+
+				if (dlg.exec() == QDialog::Accepted)
+				{
+					const std::string text = sstr(result);
+
+					if (text.size() == 32)
+					{
+						u32 index = 0;
+
+						for (auto& ch : klic._bytes)
```

Comment:
this loop should have its own function in one of the unedat or unself files, don't you think?

Reply:
Its a general action of converting 128-bit unsigned integer from text to a v128, simplified it into two 64-bit from_char actions.
If it needs a function it shouldnt be in those files specfically.

Reply:
It better was u128, but it's another story.